File: 6th Semester/Advance Programming Language/Stack/src/stack/Stack.java

package stack;

import java.util.Scanner;


public class Stack {

    public static void main(String[] args) {

        
      Scanner in = new Scanner(System.in);
      stackpushpop ob = new stackpushpop();
      
      boolean b = true ;
      while(b)
      {
          System.out.println("Choose option : ");
          System.out.println("push");
          System.out.println("pop");
          System.out.println("show");
          
          int c = in.nextInt();
          switch(c)
          {
              case 0:
                  b = false ;
                  break;
              case 1:
                  System.out.println("Enter your value ");
                  int a = in.nextInt();
                  ob.push(a);
                  
              case 2 :
                  ob.pop();
                  break;
              case 3:
                  ob.display();
                  break;
              default:
                  System.out.println("Worng input");
                  break;
          }
      }
      
    }
    
}

File: 3rd Semester/C-Lab day-08-3rd-semester/stack.cpp
#include<iostream>
using namespace std;

char stack[10], maxstk=3, top=0;

void push(int item)
{
    if (top==maxstk)
    {
        cout<<"\t\t\t\tOverflow\n";
    }
    else
    {
        top=top+1;
        stack[top]=item;
    }
}

void pop()
{
    if (top==0)
    {
        cout<<"\t\t\t\tUnderflow\n";
    }
    else
    {
        cout<<stack[top];
        cout<<" POP Done";
        top=top-1;

    }
}

void display()
{
    if(top>0)
    {
        cout<<"\n\t\t\tThis stack elements :  \n";
            for (int i = top; i > 0; --i)
            {
                cout<<stack[i]<<endl;
            }

    }

    else
        cout<<"stack is empty" ;
}

int main()
{
    int a;
    char b;
    cout<<"\t\t\t\t1 For Push\n\t\t\t\t2 For Display\n\t\t\t\t3 For POP\n\t\t\t\t4 For End\n";
    do
    {
        cout<<" \n\t\t\tPlease Enter your option: ";
        cin>>a;
        switch (a)
        {
        case 1:
            cout<<"\n\t\t\tEnter value: ";
            cin>>b;
            push(b);
            break;
        case 2:
            display();
            break;
        case 3:
            cout<<"Last value popped\n";
            pop();
            break;
        case 4:
            cout<<"Exit\n";
            break;
        default:
            cout<<"Invalid";
            break;
        }
    } while (a!=4);

    return 0;
}
